listenbrainz_spark/stats/release.py

```diff
@@ -2,29 +2,76 @@
 import time
 from listenbrainz_spark import config
 from listenbrainz_spark.stats import run_query
+from dateutil.relativedelta import relativedelta
+from datetime import datetime 
+from collections import defaultdict
 
+def get_recordings():
+    result = run_query("""
+        SELECT release_msid
+             , track_name
+             , recording_mbid
+             , recording_msid
+             , count(recording_msid) as cnt
+          FROM listen
+      GROUP BY release_msid, track_name, recording_mbid, recording_msid
+      ORDER bY cnt DESC
+    """)
+    result.show()
+    recordings = defaultdict(list)
+    for row in result.collect():
+        recordings[row.release_msid].append({
+            'track_name' : row.track_name,
+            'recording_msid' : row.recording_msid,
+            'recording_mbid' : row.recording_mbid,
+            'cnt' : row.cnt,
+        })
+    return recordings
 
-def get_most_popular():
+def get_listener():
     result = run_query("""
-        SELECT artist_msid
-             , artist_name
-             , release_msid
-             , release_name
+        SELECT release_msid
+             , user_name
              , count(release_msid) as cnt
           FROM listen
-      GROUP BY artist_msid, artist_name, release_msid, release_name
+      GROUP BY release_msid, user_name
       ORDER BY cnt DESC
     """)
     result.show()
+    listener = defaultdict(list)
+    for row in result.collect():
+        listener[row.release_msid].append({
+            'user_name': row.user_name,
+            'listen_count' : row.cnt,
+        })
+    return listener
 
-
+def get_listen_count():
+    result = run_query("""
+        SELECT release_msid
+             , release_name
+             , count(release_msid) as cnt
+          FROM listen
+      GROUP BY release_msid, release_name
+      ORDER BY cnt DESC
+    """)
+    result.show()
+    release_count = defaultdict(dict)
+    for row in result.collect():
+        release_count[row.release_msid] = {
+            'listen_count' : row.cnt,
+            'release_name' : row.release_name,
+        }
+    return release_count
 
 def main(app_name):
     t0 = time.time()
     listenbrainz_spark.init_spark_session(app_name)
     df = None
-    for y in range(2018, 2019):
-        for m in range(12, 13):
+    t = datetime.utcnow().replace(day=1)
```

Comment:
This code is duplicated a lot, we should consider creating a function that takes a time range and returns a dataframe with listens between that timeframe.

Reply:
The loop was for testing purpose. Like we have decided, stats would be calculated for the previous month/week, this code would no longer be needed.